### DISCUSSION TITLE
Application Metadata (WORK IN PROGRESS)

### DISCUSSION THREAD

**Discussion by amcgee** [dhis2/notes#146] [Specs & RFCs]
This is a **WIP** placeholder for an application metadata specification.

Currently, DHIS2 applications use a long-deprecated [Mozilla manifest.webapp](https://developer.mozilla.org/en-US/docs/Archive/B2G_OS/Firefox_OS_apps/Building_apps_for_Firefox_OS/Manifest) file to declare some metadata about the application.  This is serialized by the DHIS2 core [here](https://github.com/dhis2/dhis2-core/tree/master/dhis-2/dhis-api/src/main/java/org/hisp/dhis/appmanager/App.java)

*First thing's first* - We need to [move to the new `manifest.webapp` standard](https://jira.dhis2.org/browse/DHIS2-7933) which [supports PWAs](#376) ([w3c spec](https://w3c.github.io/manifest/))

Some metadata we currently support in the manifest:
1. Application name (for display in Application Menu)
2. Application version
3. Application description
4. Icons (notably for display in Application Menu)
4. Developer name & email
5. Application type (Web App, Dashboard Widget, Tracker Dashboard Widget) - **TO CLARIFY, see dhis2/app-platform#375**
6. Server discovery (`activities.dhis.href`) **DEPRECATED**
7. Custom application authorities
8. Several other things like `baseUrl`, `launchUrl`, `locales` supported etc. (possibly not used...)

Some metadata we currently DO NOT support but should consider (possibly outside the manifest):
1. Localized application title and description
2. Offline mode supported? dhis2/app-platform#376 
3. Inter-application dependencies
4. Minimum and maximum supported DHIS2 core versions
5. Alternative code entrypoints (such as dashboard plugin)
6. Object types supported (for Intent-like dynamic "open with" functionality)
7. "Exported" application sections (see previous Headerbar deep-linking to Maintenance App sections)

**Comment by HendrikThePendric** [comment]
> Icons (notably for display in Application Menu)

I think we need to distinguish between our custom icons for the application menu, and the icons [as described in the w3c specs](https://w3c.github.io/manifest/#icons-member).
